operatorcourier/const_io.py

```diff
@@ -0,0 +1,97 @@
+        "required" : True
+    }
+]
+
+metadata_annotations_required_fields = [
+    {
+        "field" : "description",
+        "description" : "Without this field, the description displayed in the tiles of the UI will be a truncated version of spec.description.",
+        "required" : True
+    },
+    {
+        "field" : "categories",
+        "description" : "Without this field, the operator will be categorized as Other.",
+        "required" : False
+    },
+    {
+        "field" : "capabilities",
+        "description" : "Without this field, the operator will be assigned the basic install capability - you can read more about operator maturity models here https://www.operatorhub.io/getting-started#How-do-I-start-writing-an-Operator?.",
+        "required" : True
+    },
+    {
+        "field" : "repository",
+        "description" : "Without this field, the link to the operator source code will not be displayed in the UI.",
+        "required" : False
+    },
+    {
+        "field" : "createdAt",
+        "description" : "Without this field, the time stamp at which the operator was created will not be displayed in the UI.",
+        "required" : False
+    },
+    {
+        "field" : "containerImage",
+        "description" : "Without this field, the link to the operator image will not be displayed in the UI.",
+        "required" : True
+    },
+    {
+        "field" : "alm-examples",
+        "description" : "Without this field, users will not have examples of how to write Custom Resources for the operator.",
+        "required" : False
+    }
+]
+
+spec_required_fields = [
+    {
+        "field" : "displayName",
+        "description" : "Without this field, the name of the operator for both the tile and the details view will default to metadata.name.",
+        "required" : True
+    },
+    {
+        "field" : "description",
+        "description" : "Without this field, the description in the details view of the operator will default to metadata.annotations.description.",
+        "required" : True
+    },
+    {
+        "field" : "version",
+        "description" : "Without this field, the version in the details view of the operator will be empty.",
+        "required" : True
+    },
+    {
+        "field" : "links",
+        "description" : "Without this field, no links will be displayed in the details page side panel.",
+        "required" : True
+    },
+    {
+        "field" : "icon",
+        "description" : "Without this field, the operator will display a default operator framework icon.",
+        "required" : False
+    },
+    {
+        "field" : "provider",
+        "description" : "Without this field, users will not be able to filter for provider.",
+        "required" : True
+    },
+    {
+        "field" : "maintainers",
+        "description" : "Without this field, the operator details page will not display the name and contact for users to get support in using the operator.",
```

Comment:
Could this include an example of what the field should look like? eg:
```
  maintainers:
  - name: Example
    email: example@email.com
```
There should be some mention of CSV documentation on a validation failure or warning to help people figure out how to populate these fields.